contents/code/main.js

```diff
@@ -238,27 +239,35 @@ function onDesktopSwitch(oldDesktop)
 	const oldDesktopIndex = compat.findDesktop(allDesktops, compat.toDesktop(oldDesktop));
 	const currentDesktopIndex = compat.findDesktop(allDesktops, compat.toDesktop(workspace.currentDesktop));
 	const getDesktopsLength = () => compat.workspaceDesktops().length;
-	const keepEmptyMiddleDesktops = readConfig("keepEmptyMiddleDesktops", false);
 
 	if (oldDesktopIndex <= currentDesktopIndex)
 	{
 		debug("Desktop switched to the right - ignoring");
 		return;
 	}
 
+	// Calculate the index of the last desktop we want to preserve when cleaning up.
+	// We need to preserve:
+	//   1. The current desktop (currentDesktopIndex)
+	//   2. At least MIN_DESKTOPS. Note we actually subtract two because:
+	//      - We always have the dynamic empty desktop at the end 
+	//      - MIN_DESKTOPS is a count, but we need an index
+	const preserveUpToIndex = Math.max(currentDesktopIndex, MIN_DESKTOPS - 2);
```

Comment:
I thought the purpose of #21 (and one another I can't find) was that we want to remove the leading desktop as well, for using this script with the Overview effect (the super-w thing): there you can drag a window to the plus sign and it will create a desktop

Reply:
Ah wait, I just tested and we need to disable creating the empty desktop altogether for that to work as you would expect. This constraint makes sense then